### PR TITLE
Python 3 fixes in utils.py

### PR DESCRIPTION
**Pull request by thomasst**
This makes the utils.py module importable from Python 3 apps. It also reduces required dependencies by inlining certain 3rd-party imports. `class Semaphore` backport is being removed since we don't use it / don't plan on using it.

### CODE REVIEW

File: flask_common/utils.py [hunk 1]
```diff
@@ -49,19 +47,24 @@ def json_list_generator(results):
...
-
-    def __contains__(self, key):
-        if not self:
-            return False
-        try:
-            return self[self.bisect_left(key)].lower() == key.lower()
-        except IndexError:
-            return False
+try:
+    from blist import sortedset
```
Comment (wojcikstefan) [inline flask_common/utils.py]: so we end up with either `flask_common.utils.sortedset` or `flask_common.utils.isortedset` (notice the "i" prefix in the latter case) depending on the installed packages?

Comment (thomasst) [inline flask_common/utils.py]: You can only do `from flask_common.utils import isortedset` if `blist` is installed. Previously you wouldn't be able to import the entire module. You're not supposed to import `flask_common.utils.sortedset` (even though you can).

Comment (wojcikstefan) [inline flask_common/utils.py]: ah, gotcha.

**Comment by wojcikstefan** [comment]
Looks fine at a glance

**Pull request merged**